<img src="img/usmlogo.png" style="float:right;height:100px">
<img src="img/dilogo.png" style="float:left;height:60px">
<center>
<div>
    <center>
    <h1>Laboratorio 1 Estadística Computacional</h1>
    </center>
</div>
<div>
    <br>
    <h5> Giorgio Pellizzari - 201473599-5 - Malla Nueva</h5>
    <h5> Felipe Vega - 201473511-1 - Malla Nueva</h5>
    <h6>9 de abril del 2018</h6>
</div>
</center>

In [8]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import warnings
warnings.filterwarnings('ignore')

# Simulación

En la siguiente simulación se considerará el siguiente supuesto: __Cada posible camino en una trifurcación llevará al mismo punto, donde se encuentra la siguiente trifurcación, donde se volverá a elegir que camino seguir__. En un punto, cada camino tiene la misma probabilidad de ser elegido $\left(\frac{1}{3}\right)$, independiente de la decisión tomada en el punto anterior.

El objeto del estudio será el tiempo de viaje entre las ciudades __A__ y __B__, por lo que se definirá la variable aleatoria $X$ como el tiempo que demora en llegar el conductor desde la ciudad __A__ a la __B__.

Se realizará una simulación que considerará 200000 viajes, desde los cuales se obtendrán los tiempos para cada uno. Como es necesario obtener los mismos resultados cada vez que se ejecute el bloque de código con la simulación, se utilizará un generador de números pseudo-aleatorios inicializado con una semilla arbitraria. 

El resultado de cada uno de los viajes se guardará en el arreglo __"tiemposDeViaje"__. Además, como luego se requiere calcular la probabilidad de que el conductor decida tomar el camino medio 5 veces seguidas, se mantiene el contador _**contadorDeMExitos**_ para almacenar la cantidad de veces que se da esta situación.

In [9]:
np.random.seed(2)
tiemposDeViaje = []
contadorDeMExitos = 0
for i in range(200000):
    tiempo = 0
    randomNumber = np.random.randint(3,size=20)
    cincoMContador = 0
    for i in randomNumber:
        if i == 0:
            tiempo += 6
            cincoMContador = 0
        elif i == 1:
            tiempo += 8
            cincoMContador += 1
        else:
            tiempo += 10
            cincoMContador = 0
        if cincoMContador >= 5:
            contadorDeMExitos += 1
        
    tiemposDeViaje.append(tiempo)
tiemposDeViaje = np.asarray(tiemposDeViaje)


Para obtener el valor esperado se utilizará el _**Teorema Del Límite Central**_ [1], el cual indica que si se tiene la media $(\bar{X})$ de una muestra aleatoria de un tamaño _**n**_ lo suficientemente grande, $(\bar{X})$ se puede aproximar a una distribución normal. Como en este caso se tiene un $n = 200000$ es posible aplicar el teorema, llegando a una aproximación normal, donde el valor esperado de la muestra corresponderá a la media de esta. Para calcular la media muestral se utiliza el método _**mean()**_ de _**Numpy**_.

In [10]:
valorEsperado = tiemposDeViaje.mean()
print("El valor esperado del tiempo de viaje es de: "+str(valorEsperado)+" [min]")

El valor esperado del tiempo de viaje es de: 159.98677 [min]


De esta manera, se obtiene que el valor esperado del tiempo de viaje entre A y B es de 159.98677 minutos, lo cual es aproximadamente 160 minutos.



In [11]:
valorEsperadoTeo = (1/3)*(6 + 8 + 10)*20
print(valorEsperadoTeo)

160.0


Luego, se obtiene la varianza de la variable aleatoria utilizando el método _**var()**_ de la librería _**Numpy**_.

In [12]:
varianza = tiemposDeViaje.var()
print("La varianza de los tiempos de viaje es de: " +str(varianza)+ " [min^2]")

La varianza de los tiempos de viaje es de: 53.372284967099986 [min^2]


Con lo que se obtiene una varianza muestral de 53.3722 minutos cuadrados, aproximadamente 53 minutos cuadrados.

Para obtener la probabilidad de que el viaje dure menos de 158 minutos se utiliza la definición empírica de probabilidad, calculando el número de éxitos dividido el número de experimentos.

In [13]:
contador = 0
for i in tiemposDeViaje:
    if i < 158:
        contador += 1
probabilidad158 = contador/len(tiemposDeViaje)    
print("La probabilidad de que el viaje dure menos de 158 minutos es de: "+ str(probabilidad158))

La probabilidad de que el viaje dure menos de 158 minutos es de: 0.341905


De esta manera se obtiene que la probabilidad de que un viaje dure menos de 158 minutos es de 0.3419.

Por último, se desea calcular la probabilidad de que el conductor decida elegir el camino medio 5 veces seguidas, por lo que se calcula esta probabilidad utilizando el mismo medio anterior.

In [14]:
probabilidad5Medios = contadorDeMExitos/(16*len(tiemposDeViaje))
print("La probabilidad de que se siga el camino medio 5 veces seguidas es de: "+str(probabilidad5Medios))


La probabilidad de que se siga el camino medio 5 veces seguidas es de: 0.0040734375


Al realizar el calculo se obtiene que la probabilidad de tomar el camino medio 5 veces seguidas es de aproximadamente 0.004073.

## Referencias

- [1] Devore, J. L. (2008). Probabilidad y estadística para ingenierías y ciencias. (pp 234) Cengage Learning Editores.
- [2] Pólya, G.. "Über eine Aufgabe der Wahrscheinlichkeitsrechnung betreffend die Irrfahrt im Straßennetz." Mathematische Annalen 84 (1921): 149-160. 